### PR TITLE
feat: allow passing falsy plugins

### PR DESCRIPTION
**Pull request by Asuka109**
## Summary

- Enable to pass falsy values as plugins.
- Warning when got webpack-like plugin object.

## Checklist

- [ ] Tests updated.
- [ ] Documentation updated.

### CODE REVIEW

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ export type {
   NormalizedConfig,
   // Plugin Types
   RsbuildPlugin,
+  RsbuildPlugins,
```
Comment (chenjiahan) [inline packages/core/src/index.ts:22]: why do we need to export this type in the entry module?

File: packages/shared/src/pluginStore.ts [hunk 2]
```diff
@@ -3,78 +3,74 @@ import type {
   PluginStore,
   RsbuildPlugin,
   RsbuildPluginAPI,
-  BundlerPluginInstance,
+  RsbuildPlugins,
 } from './types';
 import { color, isFunction } from './utils';
 
-function validatePlugin(plugin: unknown) {
```
Comment (chenjiahan) [inline packages/shared/src/pluginStore.ts:10]: why remove the validatePlugin function

File: packages/shared/src/pluginStore.ts [hunk 3]
```diff
@@ -3,78 +3,74 @@ import type {
...
     options?: { before?: string },
   ) => {
     const { before } = options || {};
-    newPlugins.forEach((newPlugin) => {
-      validatePlugin(newPlugin);
+    for (const plugin of newPlugins) {
+      if (!plugin) continue;
+      if (typeof plugin !== 'object') {
+        throw new Error(
+          `expect plugin instance is object, but got ${typeof plugin}.`,
```
Comment (chenjiahan) [inline packages/shared/src/pluginStore.ts:22]: the message is reverted 🌚

File: packages/core/src/index.ts (hunk 1 above)
Comment (Asuka109) [reply packages/core/src/index.ts:22]: I believe that all types used by the config object should be exported. This way, upper toolkits like modern.js can extend or assemble the config type from scratch.

File: packages/shared/src/pluginStore.ts (hunk 2 above)
Comment (Asuka109) [reply packages/shared/src/pluginStore.ts:10]: I removed it while resolving conflicts. Should I bring it back?

File: packages/core/src/index.ts (hunk 1 above)
Comment (chenjiahan) [reply packages/core/src/index.ts:22]: This file only exports **public and documented** public API, there is a comment:

image

**Pull request merged**